File: 0011-container-with-most-water/0011-container-with-most-water.rs
use std::cmp;
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        
        
        // max width
        
        let mut left = 0;
        let mut right = height.len() - 1;

        
        let mut width = (cmp::min(height[right], height[left])) * (right - left) as i32;
        
        while left < right {
            if height[left] <= height[right] {
                left += 1;
            } else {
                right -= 1;
            }

            let new_w = (cmp::min(height[right], height[left])) * (right - left) as i32;
            width = cmp::max(width, new_w);
        }
        
        
        width
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {

        int left = 0;
        int right = height.Length - 1;
        
        int max = 0;

        while(left < right) {
            
            int b = right - left;
            int h = Math.Min(height[left], height[right]);
            
            max = Math.Max(b * h, max);
            
            if(height[left] == height[right]) {
                left++;
                right--;
            }

            else if(height[left] <= height[right]) {
                left++;
            }
            else {
                right--;
            }
        }
        
        return max;
    }
}